src/components/plan/PlanMergeReview.svelte

```diff
@@ -293,19 +299,27 @@
   }
 
   async function onApproveChanges() {
-    if (initialMergeRequest !== null) {
-      const success = await effects.planMergeCommit(
-        initialMergeRequest.id,
-        initialMergeRequest.plan_snapshot_supplying_changes.plan,
-        initialMergeRequest.plan_receiving_changes,
-        user,
-      );
-      if (success) {
-        $planReadOnlyMergeRequest = false;
-        userInitiatedMergeRequestResolution = true;
-        goto(`${base}/plans/${initialPlan.id}`);
+    // this variable checks if there are any derivation groups associated with either branch. If there are, warn them of merging behavior. Otherwise, do nothing.
+    let dgWarningModal = planHasDerivationGroups
```

Comment:
It's a little difficult to discern because of the mix of promise and await logic, but I think this can maybe be rewritten as
```
    let derivationGroupDependencyConfirmation: boolean = true;
    if (planHasDerivationGroups) {
      const { confirm } = await showPlanBranchMergeDerivationGroupMessageModal(sourcePlan.name, targetPlan.name);
      derivationGroupDependencyConfirmation = confirm;
    }

    // after awaiting either a modal or nothing, proceed.
    if (initialMergeRequest !== null && derivationGroupDependencyConfirmation) {
      const success = await effects.planMergeCommit(
      ...
```

Reply:
Resolved [here](https://github.com/NASA-AMMOS/aerie-ui/commit/760387c5a994ded11c44c16adf8ae5a0c8f65275).